tests/JsonSchema/Tests/RefResolverTest.php

```diff
@@ -334,4 +334,26 @@ public function testFetchRef()
 
         $this->assertEquals($jsonSchema, $resolver->fetchRef($ref, $sourceUri));
     }
+
+    /**
+     * @expectedException \JsonSchema\Exception\JsonDecodingException
+     */
+    public function testMaxDepthExceeded()
+    {
+        // stub schema
+        $jsonSchema = new \stdClass;
+        $jsonSchema->id = 'stub';
+        $jsonSchema->additionalItems = 'stub';
+
+
+        // mock retriever
+        $retriever = $this->getMock('JsonSchema\Uri\UriRetriever', array('retrieve'));
+        $retriever->expects($this->any())->method('retrieve')->will($this->returnValue($jsonSchema));
+
+        // stub resolver
+        \JsonSchema\RefResolver::$maxDepth = 0;
+        $resolver = new \JsonSchema\RefResolver($retriever);
+
+        $this->assertEquals($jsonSchema, $resolver->resolve($jsonSchema));
```

Comment:
@f-go yes, please refactor the test and rebase to a single commit.

@Maks3w @alecsammon would you mind helping with a quick review?